storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+    // for java
+    // try (SetMockedSupervisorStormConf mocked = new SetMockedSupervisorStormConf(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorStormConf. conf)]
+    //     run test ...)
+    public static class SetMockedSupervisorStormConf implements Closeable {
+        public SetMockedSupervisorStormConf(Map conf) {
+            mockedSupervisorStormConf = conf;
+        }
+
+        @Override
+        public void close() {
+            mockedSupervisorStormConf = null;
+        }
+    }
+    private static Map mockedSupervisorStormConf = null;
+    public static Map readSupervisorStormConf(Map conf, String stormId) throws IOException {
+        if (mockedSupervisorStormConf != null) {
+            return mockedSupervisorStormConf;
+        }
+        String stormRoot = supervisorStormDistRoot(conf, stormId);
+        String confPath = supervisorStormConfPath(stormRoot);
+        return readSupervisorStormConfGivenPath(conf, confPath);
+    }
+
+    public static StormTopology readSupervisorTopology(Map conf, String stormId) throws IOException {
+        String stormRoot = supervisorStormDistRoot(conf, stormId);
+        String topologyPath = supervisorStormCodePath(stormRoot);
+        return Utils.deserialize(FileUtils.readFileToByteArray(new File(topologyPath)), StormTopology.class);
+    }
+
+    public static String workerUserRoot(Map conf) {
+        return (absoluteStormLocalDir(conf) + FILE_SEPARATOR + "/workers-users");
+    }
+
+    /* Never get used TODO : may delete it*/
+    public static String workerUserFile(Map conf, String workerId) {
+        return (workerUserRoot(conf) + FILE_SEPARATOR + workerId);
+    }
+
+    public static String getWorkerUser(Map conf, String workerId) {
+        LOG.info("GET worker-user for {}", workerId);
+        File file = new File(workerUserFile(conf, workerId));
+
+        try (InputStream in = new FileInputStream(file);
+             Reader reader = new InputStreamReader(in);
+             BufferedReader br = new BufferedReader(reader);) {
+            StringBuilder sb = new StringBuilder();
+            int r;
+            while ((r = br.read()) != -1) {
+                char ch = (char)r;
+                sb.append(ch);
+            }
+            String ret = sb.toString().trim();
+            return ret;
+        } catch (IOException e) {
+            LOG.error("Failed to get worker user for " + workerId + ".");
+            return null;
+        }
+    }
+
+    public static String getIdFromBlobKey(String key) {
+        if (key == null) return null;
+        String ret = null;
+        if (key.endsWith("-stormjar.jar")) {
+            ret = key.substring(0, key.length() - 13);
+        } else if (key.endsWith("-stormcode.ser")) {
+            ret = key.substring(0, key.length() - 14);
+        } else if (key.endsWith("-stormconf.ser")) {
+            ret = key.substring(0, key.length() - 14);
+        }
```

Comment:
This works, it is not the RE used in the clojure, but it should be fine.  My only comment would be to have static final strings for "-stormjar.jar", etc. and use the .length() of the static final string instead of the hard coded length in the substring.